server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/Quotas.scala

```diff
@@ -59,4 +68,123 @@ case class Quota(quota: Map[Quotas.Type, Value]) extends AnyVal
+case class QuotaIds(value: List[UnparsedQuotaId])
+
+case class QuotaGetRequest(accountId: AccountId,
+                           ids: Option[QuotaIds],
+                           properties: Option[Properties]) extends WithAccountId
+
+object JmapQuota {
+
+  val allProperties: Properties = Properties("id", "resourceType", "used", "limit", "scope", "name", "dataTypes", "warnLimit", "softLimit", "description")
+  val idProperty: Properties = Properties("id")
+
+  def propertiesFiltered(requestedProperties: Properties) : Properties = idProperty ++ requestedProperties
+}
+
+
+case class JmapQuota(id: Id,
+                     resourceType: ResourceType,
+                     used: UnsignedInt,
+                     limit: UnsignedInt,
+                     scope: Scope,
+                     name: QuotaName,
+                     dataTypes: List[DataType],
+                     warnLimit: Option[UnsignedInt] = None,
+                     softLimit: Option[UnsignedInt] = None,
+                     description: Option[QuotaDescription] = None)
+
+object QuotaName {
+  def from(scope: Scope, resourceType: ResourceType, dataTypes: List[DataType]): QuotaName =
+    QuotaName(s"${scope.asString()}:${resourceType.asString()}:${dataTypes.map(_.asString()).mkString("_")}")
+}
+
+case class QuotaName(string: String)
+
+case class QuotaDescription(string: String)
+
+sealed trait ResourceType {
+  def asString(): String
+}
+
+case object CountResourceType extends ResourceType {
+  override def asString(): String = "count"
+}
+
+case object OctetsResourceType extends ResourceType {
+  override def asString(): String = "octets"
+}
+
+object Scope {
+  def fromJava(scope: org.apache.james.mailbox.model.Quota.Scope): Scope =
+    scope match {
+      case org.apache.james.mailbox.model.Quota.Scope.Domain => DomainScope
+      case org.apache.james.mailbox.model.Quota.Scope.Global => GlobalScope
+      case org.apache.james.mailbox.model.Quota.Scope.User => AccountScope
+    }
+}
+
+trait Scope {
+  def asString(): String
+
+}
+
+case object AccountScope extends Scope {
+  override def asString(): String = "account"
+}
+
+case object DomainScope extends Scope {
+  override def asString(): String = "domain"
+}
+
+case object GlobalScope extends Scope {
+  override def asString(): String = "global"
+}
+
+trait DataType {
+  def asString(): String
+}
+
+case object MailDataType extends DataType {
+  override def asString(): String = "Mail"
+}
+
+case object StorageDataType extends DataType {
+  override def asString(): String = "Storage"
+}
+
+case class QuotaGetResponse(accountId: AccountId,
+                            state: UuidState,
+                            list: List[JmapQuota],
+                            notFound: QuotaNotFound)
+
+case class QuotaNotFound(value: Set[UnparsedQuotaId]) {
+  def merge(other: QuotaNotFound): QuotaNotFound = QuotaNotFound(this.value ++ other.value)
+}
+
+object QuotaIdFactory {
+  def from(quotaRoot: ModelQuotaRoot): Id =
```

Comment:
```suggestion
  def from(quotaRoot: ModelQuotaRoot, resource: ResourceType): Id =
```

This method signature would allow the QuotaId to be unique in the list ;-)